gobblin-salesforce/src/main/java/org/apache/gobblin/typedconfig/ConstraintUtil.java

```
@@ -0,0 +1,128 @@
+
+
+/**
+ * Util class for handling constrain annotation
+ * supports: IntRange, LongRange, EnumOption, StringRegex
+ */
+@Slf4j
+public class ConstraintUtil {
+  private ConstraintUtil() {
+  }
+
+  public static Object constraint(Field field, Object value, Object defaultValue) {
+    if (value == null) {
+      return defaultValue;
+    }
+    Class type = field.getType();
+    if (type.isEnum()) {
+      return getEnumValue(field, value, defaultValue);
+    }
+    switch (type.getName()) {
+      case "int": case "java.lang.Integer":
+        return getIntValue(field, value, defaultValue);
+      case "long": case "java.lang.Long":
+        return getLongValue(field, value, defaultValue);
+      case "java.lang.String":
+        return getStringValue(field, value, defaultValue);
+      case "boolean": case "java.lang.Boolean":
+        return getBooleanValue(field, value, defaultValue);
+      case "java.util.Date":
+        return getDateValue(field, value, defaultValue);
+      default:
+          throw new RuntimeException("not supported the return type: " + type.getName());
+    }
+  }
+
+  static private Object getIntValue(Field field, Object value, Object defaultValue) {
+    IntRange intRange = field.getAnnotation(IntRange.class);
+    if (intRange == null) {
+      return value;
+    }
+    int[] range = intRange.value();
+    int intValue = Integer.parseInt(value.toString());
+    if (range.length != 2 || (range.length == 2 && intValue >= range[0] && intValue <= range[1])) {
+      return value;
+    } else {
+      return defaultValue;
+    }
+  }
+  static private Object getLongValue(Field field, Object value, Object defaultValue) {
+    LongRange longRange = field.getAnnotation(LongRange.class);
+    long[] range = longRange.value();
+    if (range == null) {
+      return value;
+    }
+    long longValue = Long.parseLong(value.toString());
+    if (range.length != 2 || (range.length == 2 && longValue > range[0] && longValue < range[1])) {
```

We should raise an error instead of using the default if an out of range value is provided. This is to avoid confusion where the user thinks that the execution used the config they provide.

---

Thanks! it makes sense.